x-pack/plugins/lens/public/plugin.ts

```diff
@@ -186,6 +218,29 @@ export class LensPlugin {
       VISUALIZE_FIELD_TRIGGER,
       visualizeFieldAction(core.application)
     );
+
+    return {
+      EmbeddableComponent: getEmbeddableComponent(startDependencies.embeddable),
+      navigateToPrefilledEditor: (input: LensEmbeddableInput) => {
+        if (input.timeRange) {
+          startDependencies.data.query.timefilter.timefilter.setTime(input.timeRange);
+        }
+        const transfer = new EmbeddableStateTransfer(
+          core.application.navigateToApp,
+          core.application.currentAppId$
+        );
+        transfer.navigateToEditor('lens', {
+          path: getEditPath(undefined),
+          state: {
+            originatingApp: '',
+            valueInput: input,
+          },
+        });
+      },
```

Comment:
I think that this pre-filled editor logic is not quite right. First of all the breadcrumbs are not set. Second of all: are we expecting that users will return to the app they came from, or that the saved Lens visualization will get added to a dashboard? Right now it suggests adding to a dashboard.

Reply:
Yes, this was intended. It's possible to have a channel back to the app (using the originating app), but that's much more complex to implement correctly because the app has to somehow persist the current state, then restore it.

IMHO this is a separate feature we can work on separately once it becomes relevant - I think the flow "solution app -> Lens -> dashboard" is what makes sense here (because solution apps don't want to provide a way to persist/load custom Lens visualizations).

This is similar to how the metrics explorer integration with TSVB works right now.